readthedocs/search/api.py

```diff
@@ -167,14 +167,15 @@ def get_all_projects(self):
         main_version = self._get_version()
         main_project = self._get_project()
 
+        all_projects = [main_project]
+
         subprojects = Project.objects.filter(
             superprojects__parent_id=main_project.id,
         )
-        all_projects = []
-        for project in list(subprojects) + [main_project]:
+        for project in subprojects:
             version = (
                 Version.internal
-                .public(user=self.request.user, project=project)
+                .public(user=self.request.user, project=project, include_hidden=False)
```

Comment:
Isn't here a permission change? Before this PR it seems that the `main_project` was also checked against the permissions the user has, and now it's not because it's added to the result outside this `for`.

Reply:
Permission for the main project is already checked before everything else, users without permission to the main project can't search anything